lib/Data/ContentTypeData.php

```diff
@@ -44,4 +44,25 @@ public function addFieldDefinitionData(FieldDefinitionData $fieldDefinitionData)
     {
         $this->fieldDefinitionsData[] = $fieldDefinitionData;
     }
+
+    /**
+     * Sort $this->fieldDefinitionsData first by position, then by creation order.
```

Comment:
ID is not really equal to creation order, that is incidentally the case now with Legacy Storage Engine, but with another engine the ID could be a hash string. Hence it would be better to use `identifier`, as this won't change between storage engines (and even between export/import).